src/tracker_manager.cpp

```diff
@@ -241,16 +241,17 @@ namespace libtorrent
 	void tracker_manager::remove_request(tracker_connection const* c)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		http_conns_t::iterator i = std::find_if(m_http_conns.begin(), m_http_conns.end()
+		auto const i = std::find_if(m_http_conns.begin(), m_http_conns.end()
 			, [c] (std::shared_ptr<http_tracker_connection> const& ptr) { return ptr.get() == c; });
 		if (i != m_http_conns.end())
 		{
 			m_http_conns.erase(i);
 			return;
 		}
 
-		udp_conns_t::iterator j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
-			, [c] (udp_conns_t::value_type const& uc) { return uc.second.get() == c; });
+		auto const j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
+			, [c](std::pair<std::uint32_t, std::shared_ptr<udp_tracker_connection>> const& uc)
```

Comment:
I don't think this signature is correct. ``m_udp_conns`` is an unordered_map, so the value type will be ``std::pair<std::uint32_t const, std::shared_ptr<udp_tracker_connection>>``. it will implicitly convert to a non-const *copy* here. This is one of the benefits of referring to the type as ``::value_type``. Perhaps you could pull it out with a using statement:

``using udp_value = typename decltype(m_udp_conns)::value_type;`` (I'm not sure whether you need the ``typename`` or not)